src/util/system.h

```diff
@@ -200,6 +196,9 @@ class ArgsManager
     std::map<OptionsCategory, std::map<std::string, Arg>> m_available_args GUARDED_BY(cs_args);
     bool m_accept_any_command GUARDED_BY(cs_args){true};
     std::list<SectionInfo> m_config_sections GUARDED_BY(cs_args);
+    fs::path m_cached_blocks_path GUARDED_BY(cs_args);
+    fs::path m_cached_datadir_path GUARDED_BY(cs_args);
+    fs::path m_cached_network_datadir_path GUARDED_BY(cs_args);
```

Comment:
As these data members are cache, why do not declare them `mutable`?
It will allow to keep `const`ness of the `GetSettingsPath` and `WriteSettingsFile`.

Reply:
I have modified this.